front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -274,35 +274,42 @@ export class RetrievalConfigurationServerRunner extends BaseActionConfigurationS
     agentConfiguration: AgentConfigurationType;
     stepActions: AgentActionConfigurationType[];
   }): number {
-    const actionCount = stepActions.filter(
-      (a) => a.sId === this.actionConfiguration.sId
-    ).length;
+    const retrievalActions: RetrievalConfigurationType[] = [];
+    for (const a of stepActions) {
+      if (a.type === "retrieval_configuration") {
```

Comment:
Don't we need all actions that use citations here?

Reply:
hmmm

Reply:
indeed

Reply:
So we have this function duplicated in 2 places, I need to fix that